### PR TITLE
CE-250 Rename Codebuild shared module for Terraform init and validate

### PR DESCRIPTION
**Pull request by gds-ahine**

### CODE REVIEW

File: codebuild/codebuild_validate_terraform/code_build_validate_terraform.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          terraform init -backend="$BACKEND_VAR_FILE";
+        else
+          terraform init;
+        fi
+      - terraform plan --out=/opt/terraform_$AWS_ACCOUNT_ID.plan
+
+artifacts:
+  name: TerraformPlanArtifacts
+  files:
+    - terraform_$AWS_ACCOUNT_ID.plan
```
Comment (danjoneslf) [inline codebuild/codebuild_validate_terraform/code_build_validate_terraform.yml:28]: Have you checked that an env var here gets populated and should there be an optional name var that gets passed into this name as well so if there are multiple terraforms in the same account it names each one differently?

Comment (gds-ahine) [reply codebuild/codebuild_validate_terraform/code_build_validate_terraform.yml:28]: This is what I wasn't sure about, should a map of AWS account IDs be passed in as a variable when calling the module, and then iterate over the set using `for_each = var.aws_accounts` ?

e.g 

```
aws_accounts = {
    staging_account = {
      environment    = "test"
      aws_account_id = 103495720024
    }

    # prod_account = {
    #   environment    = "prod"
    #   aws_account_id = 779799343306
    # }
  }
```

environment variable in codebuild project which runs Terraform init & validate -:

```
environment_variable {
      name  = "AWS_ACCOUNT_ID"
      value = each.value.aws_account_id
    }
```

Comment (danjoneslf) [reply codebuild/codebuild_validate_terraform/code_build_validate_terraform.yml:28]: I wasn't really thinking about multiple accounts but good point. I was thinking about different terraform deployments. So the first one I'm thinking of is the cloudwatch config where there are 2 separate terraforms. One that gets deployed to test and prod and 1 that gets deployed into all of our accounts. 

I was just thinking if the filename should be `terraform_plan_[account]_[purpose]` where if purpose isn't specified it defaults to `main` or `default` or something.

**Review by danjoneslf — COMMENTED** [review]
I think there are 2 separate things here: 

1. Validate terraform across multiple accounts 
2. Validate multiple separate terraforms targeting the same account

**Review by danjoneslf — APPROVED** [review]
Let's get this merged and come back to if if necessary when we start using it.

**Pull request merged**